src/FileDownloader.ts

```diff
@@ -205,4 +247,34 @@ export default class FileDownloader implements IFileDownloader {
     public async deleteAllItems(context: ExtensionContext): Promise<void> {
         await rimrafAsync(FileDownloader.getDownloadsStoragePath(context));
     }
+
+    // Gets the download link for the latest release of a repo in GitHub.
+    // Returns an empty string if the file is not found.
+    // Example:
+    // https://github.com/microsoft/vscode/releases/latest
+    // owner: microsoft
+    // repo: vscode
+    // fileName: 1.8.3.zip
+    // Returns: Either a link or undefined.
+    private async getGitHubDownloadLink(
+        owner: string,
+        repository: string,
+        fileName: string
+    ): Promise<Uri | undefined> {
```

Comment:
💡 Purely a thought process here - shall we try and improve this to more valid `promise` so in case of failure it returns a proper `error` type. We can totally can do this as a separate improvement PR. Also mentioning this to get your thoughts on this.

Given the whole `Errorable` type is becoming predominant - we should think of might be extracting into some consumable source - cc @peterbom,  @hsubramanianaks (purely for the idea sake for `errorable type` ) fyi purely to cc - @gambtho

Reply:
I love this idea - captured in #51